### ISSUE TITLE
Project falls over when adding third module which is an Android library module

### ISSUE THREAD

**Issue by adil-hussain-84** [foragerr/SO-35851251-Multiproject-Artifactory-Publish#1]
Hi @foragerr, thanks for putting together this sample. Really appreciate it.

I've been looking through the sample that you put together and I see that what you've done differently to what I was doing was to apply the `maven-pubish` and `com.jfrog.artifactory` plugins inside the `allProjects` block of the project root's `build.gradle` file rather than applying it outside of this block and then to apply the `'maven-pubish'` plugin to each of the sub-modules. Looks like this works when the two sub-modules are plain old `java` modules. And this works still if I add a third plain old `java` module. But here's the problem...

If I add a third module which is not a `java` module and instead a `com.android.library` module (i.e. an Android library module) then the Gradle sync falls over with this message:

```
Cannot configure the 'publishing' extension after it has been accessed.
```

I've put together this GitHub repo which demonstrates the problem:

https://github.com/adil-hussain-84/SO-35851251-Multiproject-Artifactory-Publish

Any ideas why this wouldn't work when one of the modules an Android library module and what I could do to get it to work?

**Comment by foragerr** [comment]
The android plugin does a whole bunch of dynamic project configuration behind the scenes that the java plugin doesn't. Any of those could break the artifactory plugin in theory. However, artifactory plugin supports android-plugin projects, they even provide a sample android project with artifactory publishing, so there likely is a way to get it to work. I'll take a look at your project.

**Comment by adil-hussain-84** [comment]
I think I've got it to work! Well I've definitely got the project and all sub-modules (including the Android library) to publish to Artifactory with the `com.jfrog.artifactory` and `maven-publish` plugins. Just need to hone down on what was the crucial thing that was causing the fail earlier and I'll post an update here as soon as I've nailed it.

Many many thanks for your help and assistance throughout @foragerr! :+1:

**Comment by adil-hussain-84** [comment]
I've updated my fork of this repo here...

https://github.com/adil-hussain-84/SO-35851251-Multiproject-Artifactory-Publish

... to change the project structure to closer reflect the original Stack Overflow question (one Android library module and one Java module).

Looks like building and publishing with the `com.jfrog.artifactory` and `maven-publish` plugins is working now.

The main change I made from this repo to apply the `com.jfrog.artifactory` plugin in the `allProjects` block of the root project's `build.gradle` file and to apply the `maven-publish` plugin in the `build.gradle` file of each of the individual modules.
